apps/dashboard/src/components/primitives/form/form.tsx

```diff
@@ -120,6 +121,27 @@ const FormMessagePure = React.forwardRef<
     return null;
   }
 
+  const parseMarkdownLinks = (text: string) => {
+    if (!text.includes('[') || !text.includes(']')) return text;
+
+    const markdownLinkRegex = /\[([^\]]+)\]\(([^)"']+)(?:\s+"[^"]*")?\)/g;
+
+    const htmlContent = text.replace(markdownLinkRegex, (_, text, url) => {
+      const isValidUrl = /^(https?:\/\/|\/)[^\s<>{}\\^~[\]`]+$/i.test(url);
+      if (!isValidUrl) return text;
+
+      return `<a href="${url}" class="underline text-primary hover:text-primary/80" target="_blank" rel="noopener noreferrer nofollow">${text}</a>`;
+    });
+
+    return DOMPurify.sanitize(htmlContent, {
```

Comment:
I wonder what if we just use react markdown here? Might be a little overkill tho 🤔

Reply:
What is the use case for this? Is it rendering links in validation error messages?

Reply:
this approach make the code a bit more complicated because you need to calculate that text is the `part` in `<a>part</a>`, therefore i suggested another approach in order to simplify it